animalkingdom/src/animalkingdom/AnimalActions.java

```diff
@@ -0,0 +1,60 @@
+package animalkingdom;
+
+public abstract class AnimalActions {
+	// this creates a class AnimalActions on the 'public scope'.
+	// 'abstract' class's can define a constructor, have a 'state' associated with them.
+	// Conceptually abstract classes are good for a partial implementation of an interface.
+	// an abstract class should NOT exist without a matching 'interface'
+
+	
+	int MaxId = 0;
+	// we set a integer (int) variable (MaxId) with the value of 0.
+	// this will be used later.
+
```

Comment:
Line 10:

Don't forget to make `maxId` static - the incrementing ID's must be unique to each instance when a constructor is run, and if you don't set it to be static, `maxId` will always be set to `0` upon instantiation.

Also, you might want to add in the `private` modifier as well - as it is right now, `maxId` is accessible even outside of the `AbstractAnimal` class - to maintain encapsulation, you will want this variable to only be accessible inside this class (not even your subclasses will need to use this variable).

Lastly, it seems that there is a casing typo here - for all the other times you used this variable, the `'m'` character was in lowercase - this should be fixed so that the program can run successfully and properly.